lldb/source/Target/Process.cpp

```diff
@@ -782,6 +782,44 @@ BreakpointSiteMatchesREPLBreakpoint(const BreakpointSiteSP &bp_site_sp) {
   return false;
 }
 
+/// Returns true if reason is Trace/Breakpoint/Watchpoint/PlanComplete.
+static bool IsReasonToCheckForREPLBreakpoint(StopReason reason) {
```

Comment:
This is really testing whether this is a "lldb caused" stop or an external stop, regardless of whether this is the REPL breakpoint or not.  That's an okay API to have, but the name is confusing.  Something like "IsDebuggerCausedStop" maybe?
It might seem weird to have Instrumentation in the "non-debugger caused stops" but it really is a reaction to the program doing something wrong, so that makes sense.

Reply:
One of the reasons I had put "REPL" in the name of this function was to make it blatantly obvious that this is downstream-only code, but I can accomplish the same with a comment

Reply:
Though, now that I think of it, their callsites already make that obvious.